compiler/rustc_metadata/src/creader.rs

```diff
@@ -68,11 +68,12 @@ impl std::fmt::Debug for CStore {
 pub struct CrateLoader<'a> {
     // Immutable configuration.
     sess: &'a Session,
-    metadata_loader: Box<MetadataLoaderDyn>,
+    metadata_loader: &'a MetadataLoaderDyn,
+    definitions: ReadGuard<'a, Definitions>,
```

Comment:
Is this change necessary? Only `process_extern_crate` should read from it. In addition `definitions` isn't an immutable configuration, right? `CrateLoader` is used during macro expansion, which happens before any HIR definitions are created.